test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -292,6 +294,74 @@ public void StringSegment_EqualsString_Invalid()
             Assert.False(result);
         }
 
+        public static TheoryData<StringSegment, StringComparison> DefaultStringSegmentEqualsStringSegmentData
+        {
+            get
+            {
+                // candidate / comparer / expected result
```

Comment:
Unless you want to add variants of `StringComparison.OrdinalIgnoreCase` you don't need to pass it to the test. Also, 
// candidate / comparer ~~/ expected result~~